### PR TITLE
Updates for KES Release 2023-11-09T17-35-47Z

### LINKED ISSUE #40: KES supports a dev server for quick tests
KES supports a dev server for quick tests: https://github.com/minio/kes/releases/tag/2023-11-09T17-35-47Z

You can now start a KES server with a single command (no config, TLS cert generation etc.):

```
$ kes server --dev
Version     2023-11-09T17-35-47Z    commit=53b74e38697bc68fd88dff7a3cf431db692db9ef
Runtime     go1.21.4 darwin/arm64   compiler=gc
License     AGPLv3                  https://www.gnu.org/licenses/agpl-3.0.html
Copyright   MinIO, Inc.  2015-2023  https://min.io/

KMS         In Memory
API         · https://127.0.0.1:7373/
            · https://192.168.188.79:7373/

Docs        https://min.io/docs/kes

API Key     kes:v1:ADsGCjJoWziQ82wPUG6oHbqhhlbkajaRGP+3+JSfx5Wq
Admin       7bbffa635fc160ef8048a344a53aab54e472e5c654c6339a9cec9223301808c7
Logs        error=stderr level=INFO
            audit=stdout level=INFO

=> Server is up and running...
```

On the client:

```
$ export KES_SERVER=https://127.0.0.1:7373/
$ export KES_API_KEY=kes:v1:ADsGCjJoWziQ82wPUG6oHbqhhlbkajaRGP+3+JSfx5Wq

$ kes key ls -k
```

### PR DESCRIPTION
**Pull request by djwfyi**
- Updates `kes server` for deprecated flags and new `--dev` flag
- Updates config.yaml content
- Updates quickstart with admonition for `--dev` flag

Closes #40

### CODE REVIEW

File: content/_includes/server-config.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ tls:
   cert:     ./server.cert  # Path to the TLS certificate
   password: ""             # An optional password to decrypt the TLS private key
   
+  # Specify how/whether the KES server verifies certificates presented
+  # by clients. Valid values are "on" and "off". Defaults to off, which
+  # is recommended for most use cases.
+  auth:     ""
```
Comment (ravindk89) [inline content/_includes/server-config.md:31]: We're saying valid values are `on` or `off`, but then the provided value is `""`. Does that imply the default?

File: content/cli/kes-server/_index.md [hunk 2]
```diff
@@ -33,8 +31,78 @@ The IP address and port to use for the server.
...
+### `--config`
+
+Path to the YAML-formatted config file to use for the KES server.
+
+### `--dev`
+
+Create a development server for quick tests on `127.0.0.1:7373`.
+This flag does not require a configuration file, TLS cert generation, or other setup.
+
+Keys are ephemeral and stored in memory.
```
Comment (ravindk89) [inline content/cli/kes-server/_index.md:43]:
```suggestion
Keys are ephemeral and stored in memory.

{{< admonition title="Keys Are Lost On Restart" type="important" >}}

Ephemeral data is lost when the process restarts. This includes all encryption keys stored on KES, rendering any encrypted data as permanently unreadable.

Do not encrypt data using ephemeral keys if you are not comfortable with losing that data. 

Never use this mode in production.

{{< /admonition >}}
```

**Review by ravindk89 — APPROVED** [review]
LGTM mod nits.

File: content/cli/kes-server/_index.md (hunk 2 above)
Comment (djwfyi) [reply content/cli/kes-server/_index.md:43]: Slight tweak on wording, but otherwise 👍

File: content/_includes/server-config.md (hunk 1 above)
Comment (djwfyi) [reply content/_includes/server-config.md:31]: From https://github.com/minio/kes/blob/53b74e38697bc68fd88dff7a3cf431db692db9ef/kesconf/config.go#L257C87-L257C87, empty and `off` have the same effect.

**Review by feorlen — COMMENTED** [review]
LGTM as well 🎉

**Pull request merged**